### ISSUE TITLE
[provider] pending ghost deployments causing insufficient resources

### ISSUE THREAD

**Issue by arno01** [ovrclk/akash#1439]
Have tried akash 0.14.0 and 0.14.1-rc6.

Provider: akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0

## Having 0 active leases, 0 open bids

```
$ akash query market lease list --provider akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0 --page 1 --limit 100 -o json --state active
{"leases":[],"pagination":{"next_key":null,"total":"0"}}
```

```
$ akash query market bid list --provider akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0 --page 1 --limit 100 -o json --state open
{"bids":[],"pagination":{"next_key":null,"total":"0"}}
```

## made sure to clean manifests (just in case)

```
# kubectl get providerhosts -A
No resources found
# kubectl -n lease get manifest 
No resources found in lease namespace.
# systemctl restart akash-provider
```

## 20 pending deployments ...

pstats.sh script from https://gist.github.com/arno01/db7199a08d8bcf5d921122fb881cd480

```
$ pstats.sh https://akash-provider.nixaid.com:8443
host                                    num  num-p  cpu  cpu-a  cpu-p  mem  mem-a    mem-p  fs  fs-a     fs-p
https://akash-provider.nixaid.com:8443  0    20     0    26.05  17.7   0    68.9595  33.75  0   1578.54  87.625
```

## Akash provider logs

[akash-provider.log](https://github.com/ovrclk/akash/files/7469742/akash-provider.log)

These are the **20 ghosts**, I think:

```
root@akash-master-01:~# cat akash-provider.log | grep 'reservation requested' | cut -f5 -d=
akash10th527knwhvx8x085lw9uxxv9wmlzxf44y7n57/3328818/1/1 resources
akash15ljdqcqzhzzqg0t9320k7h90xl4wqghe3403zh/2451225/1/1 resources
akash15yd3qszmqausvzpj7n0y0e4pft2cu9rt5gccda/1346631/1/1 resources
akash14s4tlmlvy0vsps05csf8l80m7nqxmau38tehp9/3299297/1/1 resources
akash15yd3qszmqausvzpj7n0y0e4pft2cu9rt5gccda/1598194/1/1 resources
akash15yd3qszmqausvzpj7n0y0e4pft2cu9rt5gccda/1842661/1/1 resources
akash1fnc04mjln6y0y7qgkkz9nwkjane50nnjxq32yf/2885243/1/2 resources
akash14s4tlmlvy0vsps05csf8l80m7nqxmau38tehp9/3021227/1/1 resources
akash14s4tlmlvy0vsps05csf8l80m7nqxmau38tehp9/2805762/1/1 resources
akash14s4tlmlvy0vsps05csf8l80m7nqxmau38tehp9/2778040/1/1 resources
akash14s4tlmlvy0vsps05csf8l80m7nqxmau38tehp9/3202496/1/1 resources
akash15egam89a3qruarxh3xe24fuuadea0ajre85meh/2872733/1/1 resources
akash1fnc04mjln6y0y7qgkkz9nwkjane50nnjxq32yf/2779907/1/2 resources
akash15egam89a3qruarxh3xe24fuuadea0ajre85meh/1965165/1/1 resources
akash15egam89a3qruarxh3xe24fuuadea0ajre85meh/3292664/1/1 resources
akash14s4tlmlvy0vsps05csf8l80m7nqxmau38tehp9/2805770/1/1 resources
akash14s4tlmlvy0vsps05csf8l80m7nqxmau38tehp9/3289647/1/1 resources
akash19v0usdgaglcaqp2xp873q0s0t76dra4fhh6ust/3299391/1/1 resources
akash19v0usdgaglcaqp2xp873q0s0t76dra4fhh6ust/3299325/1/1 resources
akash1e4nlht6dfvet0sjyfktse7cqvjdwucg27jy3wz/3275704/1/1 resources

root@akash-master-01:~# cat akash-provider.log | grep 'reservation requested' | cut -f5 -d= |wc -l
20
```

## additional info

```
# kubectl -n akash-services describe pod $(kubectl -n akash-services get pods -l app=akash-hostname-operator -o jsonpath='{.items[0].metadata.name}') | grep Image
    Image:         ghcr.io/ovrclk/akash:0.14.1-rc6
    Image ID:      ghcr.io/ovrclk/akash@sha256:7dc73136e05c351a193432b837615ef0b6bb88601e81a277dae4c41702b736f8
root@akash-master-01:~# 

root@akash-master-01:~# kubectl get nodes -o wide
NAME                         STATUS   ROLES                  AGE     VERSION   INTERNAL-IP      EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION     CONTAINER-RUNTIME
akash-master-01.nixaid.com   Ready    control-plane,master   3d22h   v1.22.3   161.97.73.227    <none>        Ubuntu 20.04.3 LTS   5.4.0-89-generic   containerd://1.5.7
akash-master-02.nixaid.com   Ready    control-plane,master   3d22h   v1.22.3   161.97.73.228    <none>        Ubuntu 20.04.3 LTS   5.4.0-89-generic   containerd://1.5.7
akash-master-03.nixaid.com   Ready    control-plane,master   3d22h   v1.22.3   161.97.76.30     <none>        Ubuntu 20.04.3 LTS   5.4.0-89-generic   containerd://1.5.7
akash-worker-01.nixaid.com   Ready    <none>                 3d21h   v1.22.3   185.211.5.95     <none>        Ubuntu 20.04.3 LTS   5.4.0-89-generic   containerd://1.5.7
akash-worker-02.nixaid.com   Ready    <none>                 29h     v1.22.3   167.86.73.47     <none>        Ubuntu 20.04.3 LTS   5.4.0-89-generic   containerd://1.5.7
akash-worker-03.nixaid.com   Ready    <none>                 17h     v1.22.3   161.97.178.239   <none>        Ubuntu 20.04.3 LTS   5.4.0-89-generic   containerd://1.5.7
```

## kubectl describe nodes

```
root@akash-master-01:~# kubectl describe nodes
Name:               akash-master-01.nixaid.com
Roles:              control-plane,master
Labels:             beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/os=linux
                    kubernetes.io/arch=amd64
                    kubernetes.io/hostname=akash-master-01.nixaid.com
                    kubernetes.io/os=linux
                    node-role.kubernetes.io/control-plane=
                    node-role.kubernetes.io/master=
                    node.kubernetes.io/exclude-from-external-load-balancers=
Annotations:        kubeadm.alpha.kubernetes.io/cri-socket: unix:///run/containerd/containerd.sock
                    node.alpha.kubernetes.io/ttl: 0
                    projectcalico.org/IPv4Address: 161.97.73.227/32
                    projectcalico.org/IPv4IPIPTunnelAddr: 10.233.65.64
                    volumes.kubernetes.io/controller-managed-attach-detach: true
CreationTimestamp:  Sat, 30 Oct 2021 21:36:48 +0200
Taints:             node-role.kubernetes.io/master:NoSchedule
Unschedulable:      false
Lease:
  HolderIdentity:  akash-master-01.nixaid.com
  AcquireTime:     <unset>
  RenewTime:       Wed, 03 Nov 2021 18:53:01 +0100
Conditions:
  Type                 Status  LastHeartbeatTime                 LastTransitionTime                Reason                       Message
  ----                 ------  -----------------                 ------------------                ------                       -------
  NetworkUnavailable   False   Sat, 30 Oct 2021 21:52:03 +0200   Sat, 30 Oct 2021 21:52:03 +0200   CalicoIsUp                   Calico is running on this node
  MemoryPressure       False   Wed, 03 Nov 2021 18:52:49 +0100   Sat, 30 Oct 2021 21:36:43 +0200   KubeletHasSufficientMemory   kubelet has sufficient memory available
  DiskPressure         False   Wed, 03 Nov 2021 18:52:49 +0100   Sat, 30 Oct 2021 21:36:43 +0200   KubeletHasNoDiskPressure     kubelet has no disk pressure
  PIDPressure          False   Wed, 03 Nov 2021 18:52:49 +0100   Sat, 30 Oct 2021 21:36:43 +0200   KubeletHasSufficientPID      kubelet has sufficient PID available
  Ready                True    Wed, 03 Nov 2021 18:52:49 +0100   Sat, 30 Oct 2021 21:37:06 +0200   KubeletReady                 kubelet is posting ready status. AppArmor enabled
Addresses:
  InternalIP:  161.97.73.227
  Hostname:    akash-master-01.nixaid.com
Capacity:
  cpu:                4
  ephemeral-storage:  203310184Ki
  hugepages-1Gi:      0
  hugepages-2Mi:      0
  memory:             8152720Ki
  pods:               110
Allocatable:
  cpu:                4
  ephemeral-storage:  187370665265
  hugepages-1Gi:      0
  hugepages-2Mi:      0
  memory:             8050320Ki
  pods:               110
System Info:
  Machine ID:                 86a903cccac0598ed9c92e55fca2efd8
  System UUID:                f1bd9f1c-0750-4291-b02b-4faf1a3aebc6
  Boot ID:                    3438790a-9029-4634-8c95-0d3036bb9167
  Kernel Version:             5.4.0-89-generic
  OS Image:                   Ubuntu 20.04.3 LTS
  Operating System:           linux
  Architecture:               amd64
  Container Runtime Version:  containerd://1.5.7
  Kubelet Version:            v1.22.3
  Kube-Proxy Version:         v1.22.3
PodCIDR:                      10.233.64.0/24
PodCIDRs:                     10.233.64.0/24
Non-terminated Pods:          (9 in total)
  Namespace                   Name                                                  CPU Requests  CPU Limits  Memory Requests  Memory Limits  Age
  ---------                   ----                                                  ------------  ----------  ---------------  -------------  ---
  kube-system                 calico-kube-controllers-75f8f6cc59-zncsv              0 (0%)        0 (0%)      0 (0%)           0 (0%)         3d22h
  kube-system                 calico-node-8n5q8                                     250m (6%)     0 (0%)      0 (0%)           0 (0%)         3d22h
  kube-system                 coredns-78fcd69978-85qgm                              100m (2%)     0 (0%)      70Mi (0%)        170Mi (2%)     3d22h
  kube-system                 coredns-78fcd69978-r78dz                              100m (2%)     0 (0%)      70Mi (0%)        170Mi (2%)     3d22h
  kube-system                 etcd-akash-master-01.nixaid.com                       100m (2%)     0 (0%)      100Mi (1%)       0 (0%)         3d22h
  kube-system                 kube-apiserver-akash-master-01.nixaid.com             250m (6%)     0 (0%)      0 (0%)           0 (0%)         3d22h
  kube-system                 kube-controller-manager-akash-master-01.nixaid.com    200m (5%)     0 (0%)      0 (0%)           0 (0%)         3d22h
  kube-system                 kube-proxy-nw2vs                                      0 (0%)        0 (0%)      0 (0%)           0 (0%)         3d22h
  kube-system                 kube-scheduler-akash-master-01.nixaid.com             100m (2%)     0 (0%)      0 (0%)           0 (0%)         3d22h
Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  Resource           Requests     Limits
  --------           --------     ------
  cpu                1100m (27%)  0 (0%)
  memory             240Mi (3%)   340Mi (4%)
  ephemeral-storage  0 (0%)       0 (0%)
  hugepages-1Gi      0 (0%)       0 (0%)
  hugepages-2Mi      0 (0%)       0 (0%)
Events:              <none>


Name:               akash-master-02.nixaid.com
Roles:              control-plane,master
Labels:             beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/os=linux
                    kubernetes.io/arch=amd64
                    kubernetes.io/hostname=akash-master-02.nixaid.com
                    kubernetes.io/os=linux
                    node-role.kubernetes.io/control-plane=
                    node-role.kubernetes.io/master=
                    node.kubernetes.io/exclude-from-external-load-balancers=
Annotations:        kubeadm.alpha.kubernetes.io/cri-socket: /run/containerd/containerd.sock
                    node.alpha.kubernetes.io/ttl: 0
                    projectcalico.org/IPv4Address: 161.97.73.228/32
                    projectcalico.org/IPv4IPIPTunnelAddr: 10.233.85.0
                    volumes.kubernetes.io/controller-managed-attach-detach: true
CreationTimestamp:  Sat, 30 Oct 2021 21:37:42 +0200
Taints:             node-role.kubernetes.io/master:NoSchedule
Unschedulable:      false
Lease:
  HolderIdentity:  akash-master-02.nixaid.com
  AcquireTime:     <unset>
  RenewTime:       Wed, 03 Nov 2021 18:53:10 +0100
Conditions:
  Type                 Status  LastHeartbeatTime                 LastTransitionTime                Reason                       Message
  ----                 ------  -----------------                 ------------------                ------                       -------
  NetworkUnavailable   False   Sat, 30 Oct 2021 21:52:00 +0200   Sat, 30 Oct 2021 21:52:00 +0200   CalicoIsUp                   Calico is running on this node
  MemoryPressure       False   Wed, 03 Nov 2021 18:49:28 +0100   Sat, 30 Oct 2021 21:37:42 +0200   KubeletHasSufficientMemory   kubelet has sufficient memory available
  DiskPressure         False   Wed, 03 Nov 2021 18:49:28 +0100   Sat, 30 Oct 2021 21:37:42 +0200   KubeletHasNoDiskPressure     kubelet has no disk pressure
  PIDPressure          False   Wed, 03 Nov 2021 18:49:28 +0100   Sat, 30 Oct 2021 21:37:42 +0200   KubeletHasSufficientPID      kubelet has sufficient PID available
  Ready                True    Wed, 03 Nov 2021 18:49:28 +0100   Sat, 30 Oct 2021 21:37:55 +0200   KubeletReady                 kubelet is posting ready status. AppArmor enabled
Addresses:
  InternalIP:  161.97.73.228
  Hostname:    akash-master-02.nixaid.com
Capacity:
  cpu:                4
  ephemeral-storage:  203310184Ki
  hugepages-1Gi:      0
  hugepages-2Mi:      0
  memory:             8152716Ki
  pods:               110
Allocatable:
  cpu:                4
  ephemeral-storage:  187370665265
  hugepages-1Gi:      0
  hugepages-2Mi:      0
  memory:             8050316Ki
  pods:               110
System Info:
  Machine ID:                 86a903cccac0598ed9c92e55fca2efd8
  System UUID:                0960bc6d-da2d-4256-ac48-a87d9c0e80af
  Boot ID:                    4d062992-de01-42dc-b2de-b82850337bee
  Kernel Version:             5.4.0-89-generic
  OS Image:                   Ubuntu 20.04.3 LTS
  Operating System:           linux
  Architecture:               amd64
  Container Runtime Version:  containerd://1.5.7
  Kubelet Version:            v1.22.3
  Kube-Proxy Version:         v1.22.3
PodCIDR:                      10.233.65.0/24
PodCIDRs:                     10.233.65.0/24
Non-terminated Pods:          (6 in total)
  Namespace                   Name                                                  CPU Requests  CPU Limits  Memory Requests  Memory Limits  Age
  ---------                   ----                                                  ------------  ----------  ---------------  -------------  ---
  kube-system                 calico-node-lr98d                                     250m (6%)     0 (0%)      0 (0%)           0 (0%)         3d22h
  kube-system                 etcd-akash-master-02.nixaid.com                       100m (2%)     0 (0%)      100Mi (1%)       0 (0%)         3d22h
  kube-system                 kube-apiserver-akash-master-02.nixaid.com             250m (6%)     0 (0%)      0 (0%)           0 (0%)         3d22h
  kube-system                 kube-controller-manager-akash-master-02.nixaid.com    200m (5%)     0 (0%)      0 (0%)           0 (0%)         3d22h
  kube-system                 kube-proxy-js2rj                                      0 (0%)        0 (0%)      0 (0%)           0 (0%)         3d22h
  kube-system                 kube-scheduler-akash-master-02.nixaid.com             100m (2%)     0 (0%)      0 (0%)           0 (0%)         3d22h
Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  Resource           Requests    Limits
  --------           --------    ------
  cpu                900m (22%)  0 (0%)
  memory             100Mi (1%)  0 (0%)
  ephemeral-storage  0 (0%)      0 (0%)
  hugepages-1Gi      0 (0%)      0 (0%)
  hugepages-2Mi      0 (0%)      0 (0%)
Events:              <none>


Name:               akash-master-03.nixaid.com
Roles:              control-plane,master
Labels:             beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/os=linux
                    kubernetes.io/arch=amd64
                    kubernetes.io/hostname=akash-master-03.nixaid.com
                    kubernetes.io/os=linux
                    node-role.kubernetes.io/control-plane=
                    node-role.kubernetes.io/master=
                    node.kubernetes.io/exclude-from-external-load-balancers=
Annotations:        kubeadm.alpha.kubernetes.io/cri-socket: /run/containerd/containerd.sock
                    node.alpha.kubernetes.io/ttl: 0
                    projectcalico.org/IPv4Address: 161.97.76.30/32
                    projectcalico.org/IPv4IPIPTunnelAddr: 10.233.99.192
                    volumes.kubernetes.io/controller-managed-attach-detach: true
CreationTimestamp:  Sat, 30 Oct 2021 21:41:09 +0200
Taints:             node-role.kubernetes.io/master:NoSchedule
Unschedulable:      false
Lease:
  HolderIdentity:  akash-master-03.nixaid.com
  AcquireTime:     <unset>
  RenewTime:       Wed, 03 Nov 2021 18:53:02 +0100
Conditions:
  Type                 Status  LastHeartbeatTime                 LastTransitionTime                Reason                       Message
  ----                 ------  -----------------                 ------------------                ------                       -------
  NetworkUnavailable   False   Sun, 31 Oct 2021 07:06:42 +0100   Sun, 31 Oct 2021 07:06:42 +0100   CalicoIsUp                   Calico is running on this node
  MemoryPressure       False   Wed, 03 Nov 2021 18:52:13 +0100   Sun, 31 Oct 2021 07:06:34 +0100   KubeletHasSufficientMemory   kubelet has sufficient memory available
  DiskPressure         False   Wed, 03 Nov 2021 18:52:13 +0100   Sun, 31 Oct 2021 07:06:34 +0100   KubeletHasNoDiskPressure     kubelet has no disk pressure
  PIDPressure          False   Wed, 03 Nov 2021 18:52:13 +0100   Sun, 31 Oct 2021 07:06:34 +0100   KubeletHasSufficientPID      kubelet has sufficient PID available
  Ready                True    Wed, 03 Nov 2021 18:52:13 +0100   Sun, 31 Oct 2021 07:06:44 +0100   KubeletReady                 kubelet is posting ready status. AppArmor enabled
Addresses:
  InternalIP:  161.97.76.30
  Hostname:    akash-master-03.nixaid.com
Capacity:
  cpu:                4
  ephemeral-storage:  203310184Ki
  hugepages-1Gi:      0
  hugepages-2Mi:      0
  memory:             8152716Ki
  pods:               110
Allocatable:
  cpu:                4
  ephemeral-storage:  187370665265
  hugepages-1Gi:      0
  hugepages-2Mi:      0
  memory:             8050316Ki
  pods:               110
System Info:
  Machine ID:                 86a903cccac0598ed9c92e55fca2efd8
  System UUID:                b170a347-1ffb-4cde-adf0-7988914fea3d
  Boot ID:                    9ec21fac-d36d-415e-8e63-88d6c97a9a7e
  Kernel Version:             5.4.0-89-generic
  OS Image:                   Ubuntu 20.04.3 LTS
  Operating System:           linux
  Architecture:               amd64
  Container Runtime Version:  containerd://1.5.7
  Kubelet Version:            v1.22.3
  Kube-Proxy Version:         v1.22.3
PodCIDR:                      10.233.66.0/24
PodCIDRs:                     10.233.66.0/24
Non-terminated Pods:          (6 in total)
  Namespace                   Name                                                  CPU Requests  CPU Limits  Memory Requests  Memory Limits  Age
  ---------                   ----                                                  ------------  ----------  ---------------  -------------  ---
  kube-system                 calico-node-fg6b6                                     250m (6%)     0 (0%)      0 (0%)           0 (0%)         3d22h
  kube-system                 etcd-akash-master-03.nixaid.com                       100m (2%)     0 (0%)      100Mi (1%)       0 (0%)         3d22h
  kube-system                 kube-apiserver-akash-master-03.nixaid.com             250m (6%)     0 (0%)      0 (0%)           0 (0%)         3d22h
  kube-system                 kube-controller-manager-akash-master-03.nixaid.com    200m (5%)     0 (0%)      0 (0%)           0 (0%)         3d22h
  kube-system                 kube-proxy-n5fxr                                      0 (0%)        0 (0%)      0 (0%)           0 (0%)         3d22h
  kube-system                 kube-scheduler-akash-master-03.nixaid.com             100m (2%)     0 (0%)      0 (0%)           0 (0%)         3d22h
Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  Resource           Requests    Limits
  --------           --------    ------
  cpu                900m (22%)  0 (0%)
  memory             100Mi (1%)  0 (0%)
  ephemeral-storage  0 (0%)      0 (0%)
  hugepages-1Gi      0 (0%)      0 (0%)
  hugepages-2Mi      0 (0%)      0 (0%)
Events:              <none>


Name:               akash-worker-01.nixaid.com
Roles:              <none>
Labels:             akash.network/role=ingress
                    beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/os=linux
                    kubernetes.io/arch=amd64
                    kubernetes.io/hostname=akash-worker-01.nixaid.com
                    kubernetes.io/os=linux
Annotations:        kubeadm.alpha.kubernetes.io/cri-socket: /run/containerd/containerd.sock
                    node.alpha.kubernetes.io/ttl: 0
                    projectcalico.org/IPv4Address: 185.211.5.95/32
                    projectcalico.org/IPv4IPIPTunnelAddr: 10.233.77.128
                    volumes.kubernetes.io/controller-managed-attach-detach: true
CreationTimestamp:  Sat, 30 Oct 2021 22:27:05 +0200
Taints:             <none>
Unschedulable:      false
Lease:
  HolderIdentity:  akash-worker-01.nixaid.com
  AcquireTime:     <unset>
  RenewTime:       Wed, 03 Nov 2021 18:53:00 +0100
Conditions:
  Type                 Status  LastHeartbeatTime                 LastTransitionTime                Reason                       Message
  ----                 ------  -----------------                 ------------------                ------                       -------
  NetworkUnavailable   False   Sat, 30 Oct 2021 22:27:58 +0200   Sat, 30 Oct 2021 22:27:58 +0200   CalicoIsUp                   Calico is running on this node
  MemoryPressure       False   Wed, 03 Nov 2021 18:52:58 +0100   Sat, 30 Oct 2021 22:27:05 +0200   KubeletHasSufficientMemory   kubelet has sufficient memory available
  DiskPressure         False   Wed, 03 Nov 2021 18:52:58 +0100   Sat, 30 Oct 2021 22:27:05 +0200   KubeletHasNoDiskPressure     kubelet has no disk pressure
  PIDPressure          False   Wed, 03 Nov 2021 18:52:58 +0100   Sat, 30 Oct 2021 22:27:05 +0200   KubeletHasSufficientPID      kubelet has sufficient PID available
  Ready                True    Wed, 03 Nov 2021 18:52:58 +0100   Sat, 30 Oct 2021 22:27:46 +0200   KubeletReady                 kubelet is posting ready status. AppArmor enabled
Addresses:
  InternalIP:  185.211.5.95
  Hostname:    akash-worker-01.nixaid.com
Capacity:
  cpu:                6
  ephemeral-storage:  409733672Ki
  hugepages-1Gi:      0
  hugepages-2Mi:      0
  memory:             16397520Ki
  pods:               110
Allocatable:
  cpu:                6
  ephemeral-storage:  377610551490
  hugepages-1Gi:      0
  hugepages-2Mi:      0
  memory:             16295120Ki
  pods:               110
System Info:
  Machine ID:                 86a903cccac0598ed9c92e55fca2efd8
  System UUID:                a9b8cf06-6f81-40e0-8c46-81e8c35bad44
  Boot ID:                    e12ce24b-9ab8-4228-a4ea-9de7bec7be29
  Kernel Version:             5.4.0-89-generic
  OS Image:                   Ubuntu 20.04.3 LTS
  Operating System:           linux
  Architecture:               amd64
  Container Runtime Version:  containerd://1.5.7
  Kubelet Version:            v1.22.3
  Kube-Proxy Version:         v1.22.3
PodCIDR:                      10.233.67.0/24
PodCIDRs:                     10.233.67.0/24
Non-terminated Pods:          (6 in total)
  Namespace                   Name                                         CPU Requests  CPU Limits  Memory Requests  Memory Limits  Age
  ---------                   ----                                         ------------  ----------  ---------------  -------------  ---
  cert-manager                cert-manager-55658cdf68-qz9wc                0 (0%)        0 (0%)      0 (0%)           0 (0%)         44h
  cert-manager                cert-manager-cainjector-967788869-6rchj      0 (0%)        0 (0%)      0 (0%)           0 (0%)         44h
  cert-manager                cert-manager-webhook-6668fbb57d-m8kf5        0 (0%)        0 (0%)      0 (0%)           0 (0%)         44h
  ingress-nginx               ingress-nginx-controller-6b4b785659-6b6lj    100m (1%)     0 (0%)      90Mi (0%)        0 (0%)         3d21h
  kube-system                 calico-node-s5plx                            250m (4%)     0 (0%)      0 (0%)           0 (0%)         3d21h
  kube-system                 kube-proxy-5n68c                             0 (0%)        0 (0%)      0 (0%)           0 (0%)         3d21h
Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  Resource           Requests   Limits
  --------           --------   ------
  cpu                350m (5%)  0 (0%)
  memory             90Mi (0%)  0 (0%)
  ephemeral-storage  0 (0%)     0 (0%)
  hugepages-1Gi      0 (0%)     0 (0%)
  hugepages-2Mi      0 (0%)     0 (0%)
Events:              <none>


Name:               akash-worker-02.nixaid.com
Roles:              <none>
Labels:             akash.network/role=ingress
                    beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/os=linux
                    kubernetes.io/arch=amd64
                    kubernetes.io/hostname=akash-worker-02.nixaid.com
                    kubernetes.io/os=linux
Annotations:        kubeadm.alpha.kubernetes.io/cri-socket: /run/containerd/containerd.sock
                    node.alpha.kubernetes.io/ttl: 0
                    projectcalico.org/IPv4Address: 167.86.73.47/32
                    projectcalico.org/IPv4IPIPTunnelAddr: 10.233.100.0
                    volumes.kubernetes.io/controller-managed-attach-detach: true
CreationTimestamp:  Tue, 02 Nov 2021 13:34:54 +0100
Taints:             <none>
Unschedulable:      false
Lease:
  HolderIdentity:  akash-worker-02.nixaid.com
  AcquireTime:     <unset>
  RenewTime:       Wed, 03 Nov 2021 18:53:00 +0100
Conditions:
  Type                 Status  LastHeartbeatTime                 LastTransitionTime                Reason                       Message
  ----                 ------  -----------------                 ------------------                ------                       -------
  NetworkUnavailable   False   Tue, 02 Nov 2021 13:35:26 +0100   Tue, 02 Nov 2021 13:35:26 +0100   CalicoIsUp                   Calico is running on this node
  MemoryPressure       False   Wed, 03 Nov 2021 18:52:34 +0100   Tue, 02 Nov 2021 13:34:54 +0100   KubeletHasSufficientMemory   kubelet has sufficient memory available
  DiskPressure         False   Wed, 03 Nov 2021 18:52:34 +0100   Tue, 02 Nov 2021 13:34:54 +0100   KubeletHasNoDiskPressure     kubelet has no disk pressure
  PIDPressure          False   Wed, 03 Nov 2021 18:52:34 +0100   Tue, 02 Nov 2021 13:34:54 +0100   KubeletHasSufficientPID      kubelet has sufficient PID available
  Ready                True    Wed, 03 Nov 2021 18:52:34 +0100   Tue, 02 Nov 2021 13:35:14 +0100   KubeletReady                 kubelet is posting ready status. AppArmor enabled
Addresses:
  InternalIP:  167.86.73.47
  Hostname:    akash-worker-02.nixaid.com
Capacity:
  cpu:                6
  ephemeral-storage:  409733672Ki
  hugepages-1Gi:      0
  hugepages-2Mi:      0
  memory:             16397528Ki
  pods:               110
Allocatable:
  cpu:                6
  ephemeral-storage:  377610551490
  hugepages-1Gi:      0
  hugepages-2Mi:      0
  memory:             16295128Ki
  pods:               110
System Info:
  Machine ID:                 86a903cccac0598ed9c92e55fca2efd8
  System UUID:                ab1cc03e-8706-4c24-9020-ee785ca2df53
  Boot ID:                    247e4ff8-9f3e-465d-b240-1dfc4424f9e5
  Kernel Version:             5.4.0-89-generic
  OS Image:                   Ubuntu 20.04.3 LTS
  Operating System:           linux
  Architecture:               amd64
  Container Runtime Version:  containerd://1.5.7
  Kubelet Version:            v1.22.3
  Kube-Proxy Version:         v1.22.3
PodCIDR:                      10.233.68.0/24
PodCIDRs:                     10.233.68.0/24
Non-terminated Pods:          (4 in total)
  Namespace                   Name                                         CPU Requests  CPU Limits  Memory Requests  Memory Limits  Age
  ---------                   ----                                         ------------  ----------  ---------------  -------------  ---
  akash-services              akash-hostname-operator-7969d58c45-tlhds     0 (0%)        0 (0%)      0 (0%)           0 (0%)         19h
  ingress-nginx               ingress-nginx-controller-6b4b785659-6ppvb    100m (1%)     0 (0%)      90Mi (0%)        0 (0%)         28h
  kube-system                 calico-node-sjs4s                            250m (4%)     0 (0%)      0 (0%)           0 (0%)         29h
  kube-system                 kube-proxy-96k9l                             0 (0%)        0 (0%)      0 (0%)           0 (0%)         29h
Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  Resource           Requests   Limits
  --------           --------   ------
  cpu                350m (5%)  0 (0%)
  memory             90Mi (0%)  0 (0%)
  ephemeral-storage  0 (0%)     0 (0%)
  hugepages-1Gi      0 (0%)     0 (0%)
  hugepages-2Mi      0 (0%)     0 (0%)
Events:              <none>


Name:               akash-worker-03.nixaid.com
Roles:              <none>
Labels:             akash.network/role=ingress
                    beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/os=linux
                    kubernetes.io/arch=amd64
                    kubernetes.io/hostname=akash-worker-03.nixaid.com
                    kubernetes.io/os=linux
Annotations:        kubeadm.alpha.kubernetes.io/cri-socket: /run/containerd/containerd.sock
                    node.alpha.kubernetes.io/ttl: 0
                    projectcalico.org/IPv4Address: 161.97.178.239/32
                    projectcalico.org/IPv4IPIPTunnelAddr: 10.233.114.128
                    volumes.kubernetes.io/controller-managed-attach-detach: true
CreationTimestamp:  Wed, 03 Nov 2021 01:49:38 +0100
Taints:             <none>
Unschedulable:      false
Lease:
  HolderIdentity:  akash-worker-03.nixaid.com
  AcquireTime:     <unset>
  RenewTime:       Wed, 03 Nov 2021 18:53:05 +0100
Conditions:
  Type                 Status  LastHeartbeatTime                 LastTransitionTime                Reason                       Message
  ----                 ------  -----------------                 ------------------                ------                       -------
  NetworkUnavailable   False   Wed, 03 Nov 2021 01:50:09 +0100   Wed, 03 Nov 2021 01:50:09 +0100   CalicoIsUp                   Calico is running on this node
  MemoryPressure       False   Wed, 03 Nov 2021 18:49:42 +0100   Wed, 03 Nov 2021 01:49:38 +0100   KubeletHasSufficientMemory   kubelet has sufficient memory available
  DiskPressure         False   Wed, 03 Nov 2021 18:49:42 +0100   Wed, 03 Nov 2021 01:49:38 +0100   KubeletHasNoDiskPressure     kubelet has no disk pressure
  PIDPressure          False   Wed, 03 Nov 2021 18:49:42 +0100   Wed, 03 Nov 2021 01:49:38 +0100   KubeletHasSufficientPID      kubelet has sufficient PID available
  Ready                True    Wed, 03 Nov 2021 18:49:42 +0100   Wed, 03 Nov 2021 01:49:58 +0100   KubeletReady                 kubelet is posting ready status. AppArmor enabled
Addresses:
  InternalIP:  161.97.178.239
  Hostname:    akash-worker-03.nixaid.com
Capacity:
  cpu:                6
  ephemeral-storage:  409733672Ki
  hugepages-1Gi:      0
  hugepages-2Mi:      0
  memory:             16397528Ki
  pods:               110
Allocatable:
  cpu:                6
  ephemeral-storage:  377610551490
  hugepages-1Gi:      0
  hugepages-2Mi:      0
  memory:             16295128Ki
  pods:               110
System Info:
  Machine ID:                 86a903cccac0598ed9c92e55fca2efd8
  System UUID:                727b9cec-962c-427e-ae8d-0d8da61b92c6
  Boot ID:                    bd1c21ba-3db8-4447-a23f-d60f45144adb
  Kernel Version:             5.4.0-89-generic
  OS Image:                   Ubuntu 20.04.3 LTS
  Operating System:           linux
  Architecture:               amd64
  Container Runtime Version:  containerd://1.5.7
  Kubelet Version:            v1.22.3
  Kube-Proxy Version:         v1.22.3
PodCIDR:                      10.233.69.0/24
PodCIDRs:                     10.233.69.0/24
Non-terminated Pods:          (3 in total)
  Namespace                   Name                                         CPU Requests  CPU Limits  Memory Requests  Memory Limits  Age
  ---------                   ----                                         ------------  ----------  ---------------  -------------  ---
  ingress-nginx               ingress-nginx-controller-6b4b785659-dr9dg    100m (1%)     0 (0%)      90Mi (0%)        0 (0%)         17h
  kube-system                 calico-node-tgg4l                            250m (4%)     0 (0%)      0 (0%)           0 (0%)         17h
  kube-system                 kube-proxy-hv82z                             0 (0%)        0 (0%)      0 (0%)           0 (0%)         17h
Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  Resource           Requests   Limits
  --------           --------   ------
  cpu                350m (5%)  0 (0%)
  memory             90Mi (0%)  0 (0%)
  ephemeral-storage  0 (0%)     0 (0%)
  hugepages-1Gi      0 (0%)     0 (0%)
  hugepages-2Mi      0 (0%)     0 (0%)
Events:              <none>
```

## kubectl get ns & kubectl get pods -A -o wide

```
root@akash-master-01:~# kubectl get ns
NAME              STATUS   AGE
akash-services    Active   3d21h
cert-manager      Active   44h
default           Active   3d22h
ingress-nginx     Active   3d21h
kube-node-lease   Active   3d22h
kube-public       Active   3d22h
kube-system       Active   3d22h
lease             Active   3d20h

root@akash-master-01:~# kubectl get pods -A -o wide
NAMESPACE        NAME                                                 READY   STATUS      RESTARTS         AGE     IP               NODE                         NOMINATED NODE   READINESS GATES
akash-services   akash-hostname-operator-7969d58c45-tlhds             1/1     Running     0                19h     10.233.100.28    akash-worker-02.nixaid.com   <none>           <none>
cert-manager     cert-manager-55658cdf68-qz9wc                        1/1     Running     0                45h     10.233.77.158    akash-worker-01.nixaid.com   <none>           <none>
cert-manager     cert-manager-cainjector-967788869-6rchj              1/1     Running     0                45h     10.233.77.156    akash-worker-01.nixaid.com   <none>           <none>
cert-manager     cert-manager-webhook-6668fbb57d-m8kf5                1/1     Running     0                45h     10.233.77.157    akash-worker-01.nixaid.com   <none>           <none>
ingress-nginx    ingress-nginx-admission-create--1-478dl              0/1     Completed   0                3d21h   10.233.77.130    akash-worker-01.nixaid.com   <none>           <none>
ingress-nginx    ingress-nginx-admission-patch--1-fxvgp               0/1     Completed   1                3d21h   10.233.77.129    akash-worker-01.nixaid.com   <none>           <none>
ingress-nginx    ingress-nginx-controller-6b4b785659-6b6lj            1/1     Running     0                3d21h   10.233.77.131    akash-worker-01.nixaid.com   <none>           <none>
ingress-nginx    ingress-nginx-controller-6b4b785659-6ppvb            1/1     Running     0                28h     10.233.100.26    akash-worker-02.nixaid.com   <none>           <none>
ingress-nginx    ingress-nginx-controller-6b4b785659-dr9dg            1/1     Running     0                17h     10.233.114.129   akash-worker-03.nixaid.com   <none>           <none>
kube-system      calico-kube-controllers-75f8f6cc59-zncsv             1/1     Running     0                3d22h   10.233.65.66     akash-master-01.nixaid.com   <none>           <none>
kube-system      calico-node-8n5q8                                    1/1     Running     0                3d22h   161.97.73.227    akash-master-01.nixaid.com   <none>           <none>
kube-system      calico-node-fg6b6                                    1/1     Running     1 (3d11h ago)    3d22h   161.97.76.30     akash-master-03.nixaid.com   <none>           <none>
kube-system      calico-node-lr98d                                    1/1     Running     0                3d22h   161.97.73.228    akash-master-02.nixaid.com   <none>           <none>
kube-system      calico-node-s5plx                                    1/1     Running     0                3d21h   185.211.5.95     akash-worker-01.nixaid.com   <none>           <none>
kube-system      calico-node-sjs4s                                    1/1     Running     0                29h     167.86.73.47     akash-worker-02.nixaid.com   <none>           <none>
kube-system      calico-node-tgg4l                                    1/1     Running     0                17h     161.97.178.239   akash-worker-03.nixaid.com   <none>           <none>
kube-system      coredns-78fcd69978-85qgm                             1/1     Running     0                3d22h   10.233.65.65     akash-master-01.nixaid.com   <none>           <none>
kube-system      coredns-78fcd69978-r78dz                             1/1     Running     0                3d22h   10.233.65.67     akash-master-01.nixaid.com   <none>           <none>
kube-system      etcd-akash-master-01.nixaid.com                      1/1     Running     3                3d22h   161.97.73.227    akash-master-01.nixaid.com   <none>           <none>
kube-system      etcd-akash-master-02.nixaid.com                      1/1     Running     0                3d22h   161.97.73.228    akash-master-02.nixaid.com   <none>           <none>
kube-system      etcd-akash-master-03.nixaid.com                      1/1     Running     21 (3d10h ago)   3d22h   161.97.76.30     akash-master-03.nixaid.com   <none>           <none>
kube-system      kube-apiserver-akash-master-01.nixaid.com            1/1     Running     3                3d22h   161.97.73.227    akash-master-01.nixaid.com   <none>           <none>
kube-system      kube-apiserver-akash-master-02.nixaid.com            1/1     Running     0                3d22h   161.97.73.228    akash-master-02.nixaid.com   <none>           <none>
kube-system      kube-apiserver-akash-master-03.nixaid.com            1/1     Running     21 (3d10h ago)   3d22h   161.97.76.30     akash-master-03.nixaid.com   <none>           <none>
kube-system      kube-controller-manager-akash-master-01.nixaid.com   1/1     Running     5 (3d22h ago)    3d22h   161.97.73.227    akash-master-01.nixaid.com   <none>           <none>
kube-system      kube-controller-manager-akash-master-02.nixaid.com   1/1     Running     0                3d22h   161.97.73.228    akash-master-02.nixaid.com   <none>           <none>
kube-system      kube-controller-manager-akash-master-03.nixaid.com   1/1     Running     1 (3d11h ago)    3d22h   161.97.76.30     akash-master-03.nixaid.com   <none>           <none>
kube-system      kube-proxy-5n68c                                     1/1     Running     0                3d21h   185.211.5.95     akash-worker-01.nixaid.com   <none>           <none>
kube-system      kube-proxy-96k9l                                     1/1     Running     0                29h     167.86.73.47     akash-worker-02.nixaid.com   <none>           <none>
kube-system      kube-proxy-hv82z                                     1/1     Running     0                17h     161.97.178.239   akash-worker-03.nixaid.com   <none>           <none>
kube-system      kube-proxy-js2rj                                     1/1     Running     0                3d22h   161.97.73.228    akash-master-02.nixaid.com   <none>           <none>
kube-system      kube-proxy-n5fxr                                     1/1     Running     1 (3d11h ago)    3d22h   161.97.76.30     akash-master-03.nixaid.com   <none>           <none>
kube-system      kube-proxy-nw2vs                                     1/1     Running     0                3d22h   161.97.73.227    akash-master-01.nixaid.com   <none>           <none>
kube-system      kube-scheduler-akash-master-01.nixaid.com            1/1     Running     5 (3d22h ago)    3d22h   161.97.73.227    akash-master-01.nixaid.com   <none>           <none>
kube-system      kube-scheduler-akash-master-02.nixaid.com            1/1     Running     0                3d22h   161.97.73.228    akash-master-02.nixaid.com   <none>           <none>
kube-system      kube-scheduler-akash-master-03.nixaid.com            1/1     Running     1 (3d11h ago)    3d22h   161.97.76.30     akash-master-03.nixaid.com   <none>           <none>
```

## Related

Probably related https://github.com/ovrclk/akash/issues/1421

**Comment by arno01** [comment]
cc @boz @dmikey

**Comment by arno01** [comment]
Yeah, these are the ghosts..
When stopping the Akash provider it is trying to close those 20 ghosts, but they fail with `failed to execute message; message index: 0: unknown lease for bid` message.


```
ov 03 20:29:37 akash-master-01.nixaid.com start-provider.sh[3662562]: I[2021-11-03|20:29:37.364] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: 2B07C00E171414BA84D345B594E66329799076FA4A4A61884202854D331CD49B\n  Raw Log: []\n  Logs: []" err=null
Nov 03 20:29:37 akash-master-01.nixaid.com start-provider.sh[3662562]: I[2021-11-03|20:29:37.428] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: 613CDB92D786F83A574E169E88E40A6EAD823C45F547C882FE8F7D770214AB44\n  Raw Log: []\n  Logs: []" err=null
Nov 03 20:29:37 akash-master-01.nixaid.com start-provider.sh[3662562]: I[2021-11-03|20:29:37.482] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: 51D7CA0002A0E3A472687CD279281034388C8D235D5826BDCA325FE12BDBADE8\n  Raw Log: []\n  Logs: []" err=null
Nov 03 20:29:37 akash-master-01.nixaid.com start-provider.sh[3662562]: I[2021-11-03|20:29:37.549] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: F47DF7FB20A37594703D5617BA4323F6DF725D61D7B00A8B3A29F41A7BE750A0\n  Raw Log: []\n  Logs: []" err=null
Nov 03 20:29:37 akash-master-01.nixaid.com start-provider.sh[3662562]: I[2021-11-03|20:29:37.605] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: FAA21EE94CEFD50C7406208753D8A0E8030BD14508955B2A7EC11401328207B0\n  Raw Log: []\n  Logs: []" err=null
Nov 03 20:29:37 akash-master-01.nixaid.com start-provider.sh[3662562]: I[2021-11-03|20:29:37.660] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: F7B4FFA70F0D51FDEA520E3F1E25AA060F4A9A126A7C698FDDBD6F7AEA4E7FFF\n  Raw Log: []\n  Logs: []" err=null
Nov 03 20:29:37 akash-master-01.nixaid.com start-provider.sh[3662562]: I[2021-11-03|20:29:37.727] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: C0D925AF65BCF6DAEAC0EC15F62CDF61D3F5200C1018DF49F75D66AA5332DB2E\n  Raw Log: []\n  Logs: []" err=null
Nov 03 20:29:37 akash-master-01.nixaid.com start-provider.sh[3662562]: I[2021-11-03|20:29:37.789] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: 5A89E7B88CFD37BCD2DC6441DD6F6F14D78CE73B3C1F1F6185153C2631F48817\n  Raw Log: []\n  Logs: []" err=null
Nov 03 20:29:37 akash-master-01.nixaid.com start-provider.sh[3662562]: I[2021-11-03|20:29:37.850] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: 8B087549DCE858647EA590B3DEFD852D6030817E5B5E07B6E736D83BC5CFB51F\n  Raw Log: []\n  Logs: []" err=null
Nov 03 20:29:37 akash-master-01.nixaid.com start-provider.sh[3662562]: I[2021-11-03|20:29:37.913] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: B6549BD6121681A3CF3F7839A316EB38932AD97E2C3A5E750FD681CA2485758F\n  Raw Log: []\n  Logs: []" err=null
Nov 03 20:29:37 akash-master-01.nixaid.com start-provider.sh[3662562]: I[2021-11-03|20:29:37.974] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: 23DFD2B3621F393E9734355CD851491EE2C15026DDD350ED3D1F05A313A5B30F\n  Raw Log: []\n  Logs: []" err=null
Nov 03 20:29:38 akash-master-01.nixaid.com start-provider.sh[3662562]: I[2021-11-03|20:29:38.036] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: 2C6E1EB12A5F11B4C539DB0CCDCA82C085128B69A684395D07AEF4A2EA17652B\n  Raw Log: []\n  Logs: []" err=null
Nov 03 20:29:38 akash-master-01.nixaid.com start-provider.sh[3662562]: I[2021-11-03|20:29:38.091] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: C6D32EC786A9485FF510A9A401C0599292EEA59F74F7A9494411AF0986D27D06\n  Raw Log: []\n  Logs: []" err=null
Nov 03 20:29:38 akash-master-01.nixaid.com start-provider.sh[3662562]: I[2021-11-03|20:29:38.147] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: 29EADBE2EA91F51F2F93AC55F4CE781D53E0CA323C245BB6453A14A26F0D9046\n  Raw Log: []\n  Logs: []" err=null

Nov 03 20:29:38 akash-master-01.nixaid.com start-provider.sh[3662562]: I[2021-11-03|20:29:38.211] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: DCB2278DE985D1BB2EF7CB022E25718CB15A30A9FD6B8BDCBF6A2A8F1148B593\n  Raw Log: []\n  Logs: []" err=null
Nov 03 20:29:38 akash-master-01.nixaid.com start-provider.sh[3662562]: I[2021-11-03|20:29:38.273] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: DB50FB7FA7C0B0C7538FD56E28E308EC3FE2E9A3B1B1B4C65682C82B4DCD261D\n  Raw Log: []\n  Logs: []" err=null
Nov 03 20:29:38 akash-master-01.nixaid.com start-provider.sh[3662562]: I[2021-11-03|20:29:38.326] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: 64C4F717467D318DB540A572CB04DB3B9199B364246F89B4EAA137F34D0C4E7F\n  Raw Log: []\n  Logs: []" err=null
Nov 03 20:29:38 akash-master-01.nixaid.com start-provider.sh[3662562]: I[2021-11-03|20:29:38.390] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: 82531614ABD019E6DD3237E8DA4AD16D5F1DAC44CF5A490873E90D404AEA0BA3\n  Raw Log: []\n  Logs: []" err=null

Nov 03 20:29:38 akash-master-01.nixaid.com start-provider.sh[3662562]: I[2021-11-03|20:29:38.448] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: 971365D28416C1A5AEB5B871B0A326820BC5243B5EEB64846D96561753A3DE18\n  Raw Log: []\n  Logs: []" err=null
Nov 03 20:29:38 akash-master-01.nixaid.com start-provider.sh[3662562]: I[2021-11-03|20:29:38.511] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: 5DEDD82E8256092A541CDD427C74BBD617E9B2FB9AE38F1CC2C52E3FC2C49A98\n  Raw Log: []\n  Logs: []" err=null
Nov 03 20:29:38 akash-master-01.nixaid.com start-provider.sh[3662562]: I[2021-11-03|20:29:38.511] Waiting on provider attributes service       module=bidengine-service
Nov 03 20:29:38 akash-master-01.nixaid.com systemd[1]: akash-provider.service: Succeeded.
Nov 03 20:29:38 akash-master-01.nixaid.com systemd[1]: Stopped Akash Provider.
```

**Comment by arno01** [comment]
Looks like all providers are affected. I have been in talk with Chiroxo, he's having the same issue:

## Chiroxo's logs when he stops his akash provider

```
Nov 03 23:59:59 vmi567667.contaboserver.net start-provider.sh[178039]: I[2021-11-03|23:59:59.546] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: AD8BACA69B94C4A25302F6B8D77999BFA31132A1B99406EE88DDAF4E5414051B\n  Raw Log: []\n  Logs: []" err=null
Nov 03 23:59:59 vmi567667.contaboserver.net start-provider.sh[178039]: I[2021-11-03|23:59:59.606] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: 90B2CA847026DD52A0F1F8F022D5FF99ACA9370D5E76AB51BFFEC479CACE57B4\n  Raw Log: []\n  Logs: []" err=null
Nov 03 23:59:59 vmi567667.contaboserver.net start-provider.sh[178039]: I[2021-11-03|23:59:59.657] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: 12AFBBA891F1C4E7C6AE429CF53C99B921D4BE3594FA636C9D59C0F2DEFF0A53\n  Raw Log: []\n  Logs: []" err=null
Nov 03 23:59:59 vmi567667.contaboserver.net start-provider.sh[178039]: I[2021-11-03|23:59:59.708] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: FA0334463F50206FB74BACF2264F21A55E43E3AEA5FBE7CDEF31D074B4C0B799\n  Raw Log: []\n  Logs: []" err=null
Nov 03 23:59:59 vmi567667.contaboserver.net start-provider.sh[178039]: I[2021-11-03|23:59:59.766] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: B0041593486A5F5413A993AD9F047CD5A79A95B15D2E5E1A2D8C5FA44A89F30F\n  Raw Log: []\n  Logs: []" err=null
```

Same error: `failed to execute message; message index: 0: unknown lease for bid`


## All active providers seem to be having the same issue

```
$ akash query provider list -o json | jq -r '.providers[] | [ .host_uri, .owner ] | @sh' | xargs -P30 -n2 sh -c 'curl -m1 -o /dev/null -skw "%{http_code}" $0/status | grep -qw 200 && echo $0' | xargs -r -P30 -I@ sh -c "pstats.sh @" | column -t
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://akash-provider.nixaid.com:8443                0    21     0       22.05    21.7   0        68.4595  34.25    0        1578.29  87.875
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://akash01.bridgetower-dc.org:8443               0    13     0       36.525   13.5   0        73.08    28.5     0        436.236  71.5
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://akash01.swiss01.bridgetower-dc.org:8443       0    13     0       36.525   13.5   0        73.08    28.5     0        436.236  71.5
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://sia25.cosow.com:8443                          0    5      0       45.275   8      0        365.859  22       0        4544.06  626
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://sia22.cosow.com:8443                          0    6      0       22.55    8.5    0        185.676  18.5     0        2580.73  28.5
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://provider.karukera-mining.com:8443             0    7      0       5.45     5.4    0        15.1798  14       0        389.752  9
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://provider.aksh.bv.cubus.ro:8443                0    8      0       68.9     1.552  0        84.9696  4.35714  0        1084.36  219.5
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://provider.provider-0.prod.ams1.akash.pub:8443  196  55     597.26  505.865  78.3   206.266  5848.53  109.25   692.766  4599.54  535.938
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://internal.ingotstudios.com:8443                3    6      3.5     18.3     5.3    3.5      38.078   15.125   5.5      2464.74  222.125
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://akashprovider.theminershive.com:8443          0    0      0       22.775   0      0        62.3863  0        0        441.592  0
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://supernaut.ddns.net:8443                       0    14     0       93.425   16.8   0        106.829  39.625   0        2468.46  670.625
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://akash.pogchia.world:8443                      0    10     0       21.8     8      0        17.988   17       0        1262.16  28
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://provider.provider-2.prod.ewr1.akash.pub:8443  133  68     284.3   1562.4   117.3  133.875  9913.3   135.25   444.813  10602.4  769.938
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://akash.pogchia.world:8443                      0    10     0       21.8     8      0        17.988   17       0        1262.16  28
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://supernaut.ddns.net:8443                       0    14     0       93.425   16.8   0        106.829  39.625   0        2468.46  670.625
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://provider.ewr1p0.mainnet.akashian.io:8443      73   0      422.7   116.672  0      172      1267.02  0        572.75   1028.38  0
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://earthlab.handyhost.computer:8443              0    10     0       21.8     8      0        17.988   17       0        1262.16  28
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://earthlab.handyhost.computer:8443              0    10     0       21.8     8      0        17.988   17       0        1262.16  28
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://stanzo.handyhost.computer:8443                4    9      4.9     16.9     5.1    5.25     12.7382  9.125    9.25     1252.91  26.125
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://provider.sm1.smartnodes.cloud:8443            1    12     1.5     43.775   14.7   1.5      224.304  35.5     2.5      11142.2  669.5
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://earthlab.handyhost.computer:8443              0    10     0       21.8     8      0        17.988   17       0        1262.16  28
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://earthlab.handyhost.computer:8443              0    10     0       21.8     8      0        17.988   17       0        1262.16  28
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://provider.sjc1p0.mainnet.akashian.io:8443      68   6      121.5   90.315   4.534  133.531  515.983  21.75    169.55   1407.25  52.625
host                                                  num  num-p  cpu     cpu-a    cpu-p  mem      mem-a    mem-p    fs       fs-a     fs-p
https://provider.provider-0.prod.sjc1.akash.pub:8443  124  58     207.7   410.95   94.3   103.324  3191.02  92.25    181.695  4426.38  636.938
```

**Comment by arno01** [comment]
workaround that works for me right now:

```
$ git diff
diff --git a/provider/bidengine/order.go b/provider/bidengine/order.go
index 886f5bb2..7f8ddba3 100644
--- a/provider/bidengine/order.go
+++ b/provider/bidengine/order.go
@@ -300,7 +300,7 @@ loop:
                        if o.bidPlaced {
                                o.log.Info("Fulfillment already exists")
                                // fulfillment already created (state recovered via queryExistingOrders)
-                               break
+                               break loop
                        }
                        pricech = runner.Do(metricsutils.ObserveRunner(func() runner.Result {
                                // Calculate price & bid
```

```
$ pstats.sh https://akash-provider.nixaid.com:8443
host                                    num  num-p  cpu  cpu-a  cpu-p  mem  mem-a    mem-p  fs  fs-a     fs-p
https://akash-provider.nixaid.com:8443  1    1      1    25.05  1      1    67.9595  2      1   1577.54  20
```

Though, provider is still trying to close the "catchup" orders, e.g.:

```
Nov 08 23:34:58 akash-master-01.nixaid.com start-provider.sh[3807667]: I[2021-11-08|23:34:58.698] broadcast response                           cmp=client/broadcaster response="Response:\n  TxHash: E26FB7FE40B326C8392E07150C99012A3F870117BB5B5307546EA9BD559FADA0\n  Raw Log: []\n  Logs: []" err=null
```

https://www.mintscan.io/akash/txs/3E12A6C607FC3D542D57D6A5D5F2D364B78899B9E1A5DF25A6AA7C86E2D1256B

still getting `failed to execute message; message index: 0: unknown lease for bid` for all 20 bids though

**Comment by hydrogen18** [comment]
@arno01 I agree that should be `break loop`, I'll see about adding a test for that and making the change.

**Comment by hydrogen18** [comment]
Thinking about this some more, I think the order manager doesn't use `break loop` there because it wants to wait for events. It needs to create `event.LeaseWon` if the bid is accepted.

**Comment by hydrogen18** [comment]
'reservation requested' is logged whenever a reservation is requested. Not when a reservation is made. It's almost irrelevant, as the provider is continously checking new orders on the blockchain to see if they
can be reserved

When you see something like this a reservation has actually been made in the available inventory

```
Nov 03 18:43:29 akash-master-01.nixaid.com start-provider.sh[3630261]: I[2021-11-03|18:43:29.718] Reservation fulfilled                        module=bidengine-order order=akash10th527knwhvx8x085lw9uxxv9wmlzxf44y7n57/3328818/1/1
```

When you see something like this it means the provider believes there is a bid on the chain

```
Nov 03 18:43:29 akash-master-01.nixaid.com start-provider.sh[3630261]: I[2021-11-03|18:43:29.718] Fulfillment already exists                   module=bidengine-order order=akash10th527knwhvx8x085lw9uxxv9wmlzxf44y7n57/3328818/1/1
```

The provider has bids on the block chain that are currently out there and no one has created a lease for them. Presently, a bid that exists on the blockchain _on startup_ is never automatically closed. My suggestion
would to be stop the provider, manually issue transactions to close unwanted bids, then start your provider back up.

Transactions such as this one https://www.mintscan.io/akash/txs/2B07C00E171414BA84D345B594E66329799076FA4A4A61884202854D331CD49B are from a bug in the provider where it issues a transaction to close a bid which does not exist at that time.

That likely is coming from here

https://github.com/ovrclk/akash/blob/mainnet/main/provider/bidengine/order.go#L381

_losing_ bids are automatically closed by the network.

https://github.com/ovrclk/akash/blob/mainnet/main/x/market/handler/server.go#L214

Submitting this TX in that case is a waste of transaction fees and network capacity but otherwise not a technical problem at this time. You can safely ignore this problem for now.

I'm going to propose a separate PR with a number of changes to address this and some other issues.

Please don't refer to these as ghost deployments. The provider sees active bids on the blockchain and tries to act on them accordingly. The blockchain has a very high duration for the maximum time a bid can remaining pending presently.
Exactly when a provider may choose to close pending bids is sort of open question. There could be lots of strategies I can think of around this. For now we're going to stick to the strictly time out based strategy as it is the simplest
one to act upon.

**Comment by arno01** [comment]
@hydrogen18 thank you for explaining the nature of the issue and addressing it!
I've compiled akash off of https://github.com/ovrclk/akash/pull/1456 -- problems seem to have resolved now! :tada:

P.S.
I called those as "ghost deployments" since that's what I've seen (or I thought I've seen 🤔) over `provider:8443/status` - never closing pending deployments [reservations] (`cluster.inventory.pending`) which were holding up the resources, without the payment [since no lease was created for those reservations], in a way that my provider could not accommodate the new ones.
There might be a better wording for these than the "ghost deployments", probably "dangling stale reservations" would sound better? 🤷‍♂️

**Comment by arno01** [comment]
There seem to be a different issue now - whenever I'm closing the deployment, the `cluster.inventory.active` is not decrementing.
Let's discuss that issue here https://github.com/ovrclk/akash/pull/1456#issuecomment-976336487

**Comment by hydrogen18** [comment]
Closing, since the main issue is resolved
